src/main/java/reminderebot/commands/DeleteCommand.java

```diff
@@ -0,0 +1,52 @@
+package reminderebot.commands;
+
+import reminderebot.TaskList;
+import reminderebot.Ui;
+import reminderebot.Storage;
+import reminderebot.ReminderebotException;
+import reminderebot.task.Task;
+import reminderebot.task.ToDo;
+import reminderebot.task.Deadline;
+import reminderebot.task.Event;
+
+
+/**
+ * The DeleteCommand class represents a command to delete a task in tasklist.
+ */
+public class DeleteCommand extends Command {
+    private final int index;
+
+    /**
+     * Create a Delete command.
+     * @param index
+     */
+    public DeleteCommand(int index) {
+        this.index = index;
+    }
+
+    /**
+     * Delete task.
+     * @param tasklist
+     * @param ui
+     * @param storage
+     * @throws ReminderebotException
+     */
+    @Override
+    public void execute(TaskList tasklist, Ui ui, Storage storage) throws ReminderebotException {
+        if (index > tasklist.length() || index < 1) { // index is out of bounds
+            throw new ReminderebotException("Item selected to be deleted is not in list.\n" +
+                    "Syntax: delete <int>");
+        }
+        Task task = tasklist.deleteTask(index);
+        ui.deleteTask(task, tasklist.length());
+    }
+
+    /**
+     * Delete command does not exit.
+     * @return false
+     */
+    @Override
+    public boolean isExit() {
+        return false;
```

Comment:
Overall the code was quite well written. Nice use of OOP principles here that led to it being easy to understand.